### DISCUSSION TITLE
Pandas dataframe to parquet. Save to S3 using S3_hook

### DISCUSSION THREAD

**Discussion by viven25** [apache/airflow#25473] [Q&A]
I currently have a working setup of Airflow in a EC2.

My goal is to save a pandas dataframe to S3 bucket in parquet format. For some unknown reason, only 0Bytes get written. 
This behavious is unexpected because i even log the contents in the dataframe.

Here is my code

```
    import io
    from airflow.providers.amazon.aws.hooks.s3 import S3Hook
    import pandas

    df: pd.Dataframe =  ...
    key = ....

    buffer = io.BytesIO()
    df.to_parquet(buffer, index=False)
    s3_hook.load_file_obj(buffer, key, cfg['bucket_name'], False)
```

Im currently using `Pyarrow 7.0.0`

<img width="1375" alt="Screenshot 2022-08-02 at 6 49 57 AM" src="https://user-images.githubusercontent.com/108452347/182384992-43f885cb-e42b-49dd-8c10-b4f4046f47f8.png">

**Comment by Taragolis** [comment]
Personal suggestion, in case if you required both Pandas and AWS better look at [aws-data-wrangler](https://aws-data-wrangler.readthedocs.io/en/2.16.1/stubs/awswrangler.s3.to_parquet.html#awswrangler-s3-to-parquet)

```python
import awswrangler as wr
import pandas as pd
from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook

df: pd.DataFrame =  ...
key = ....
hook = AwsBaseHook(aws_conn_id="your-conn-id")

# upload to S3 bucket
wr.s3.to_parquet(df=df, path=f"s3://{cfg['bucket_name']}/{key}", boto3_session=hook.get_session())

```

However if you want use S3Hook methods, than you need [change the stream position](https://docs.python.org/3/library/io.html#io.IOBase.seek) of your buffer to the beginning before upload

```python
...
buffer.seek(0, 0)
s3_hook.load_file_obj(buffer, key, cfg['bucket_name'], False)
```
